src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,39 @@
 
 class Animal {
   // write your code here
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = !this.hidden;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden) {
+      animal.health -= 50;
+
+      if (animal.health <= 0) {
+        Animal.alive = Animal.alive.filter((item) => item.health > 0);
```

Comment:
The code is checking and removing dead animals only when a Carnivore bites a Herbivore. This could potentially leave dead animals in the 'alive' array if the animal's health reaches zero by some other means.